src/base/server_type.cpp

```diff
@@ -18,17 +18,63 @@
 
 #include "server_type.hpp"
 
+#include <string>
 #include "system_parameter.h"
 
 static SERVER_TYPE g_server_type;
 
+void init_page_server_hosts ();
+
 void init_server_type ()
 {
   g_server_type = (SERVER_TYPE) prm_get_integer_value (PRM_ID_SERVER_TYPE);
+  if (g_server_type == SERVER_TYPE_TRANSACTION)
+    {
+      init_page_server_hosts ();
+    }
 }
 
 SERVER_TYPE get_server_type ()
 {
   return g_server_type;
 }
 
+void init_page_server_hosts ()
+{
+  assert (g_server_type == SERVER_TYPE_TRANSACTION);
+  std::string hosts = prm_get_string_value (PRM_ID_PAGE_SERVER_HOSTS);
+
+  if (!hosts.length ())
+    {
+      return;
+    }
+
+  auto col_pos = hosts.find (":");
+
+  if (col_pos < 1 || col_pos >= hosts.length () - 1)
+    {
+      er_set (ER_WARNING_SEVERITY, ARG_FILE_LINE, ER_HOST_PORT_PARAMETER, 2, prm_get_name (PRM_ID_PAGE_SERVER_HOSTS),
+	      hosts.c_str ());
+      return;
+    }
+
+  long port = -1;
+  try
+    {
+      port = std::stol (hosts.substr (col_pos+1));
```

Comment:
Is a string like 3000a treated as 3000?

Reply:
Yes, trailing letters are ignored by `std::stol`. I can't imagine how this could lead to errors tough. Maybe someone typing `300O` instead of `3000`...

Of course, I can parse the string if you think it's important.